### PR TITLE
Update 0.2

### PR DESCRIPTION
**Pull request by pauliesnug**
## Description
update 0.2 (see internal docs for more detials)

## Checklist
- [ ] I made a clear description of what was changed
- [ ] I stated why these changes were necessary
- [ ] I updated documentation or said what needs to be updated
- [ ] I made sure these changes are backwards compatible
- [ ] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by Wyvest — APPROVED** [review]

**Review by CaledonianEH — APPROVED** [review]
Should pass QC

**Review by xtrm-en — APPROVED** [review]
lgtm

File: src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java [hunk 1]
```diff
@@ -44,17 +46,21 @@ public class OptionSubcategory {
...
+        this.category = category;
     }
 
     public int draw(long vg, int x, int y, InputHandler inputHandler) {
         NanoVGHelper nanoVGHelper = NanoVGHelper.INSTANCE;
         String filter = OneConfigGui.INSTANCE == null ? "" : OneConfigGui.INSTANCE.getSearchValue().toLowerCase().trim();
-        filteredOptions = options.stream().filter(option -> !option.isHidden() && (filter.equals("") || name.toLowerCase().contains(filter) || option.name.toLowerCase().contains(filter))).collect(Collectors.toList());
-        List<ConfigPageButton> filteredTop = topButtons.stream().filter(page -> !page.isHidden() && (filter.equals("") || name.toLowerCase().contains(filter) || page.name.toLowerCase().contains(filter) || page.description.toLowerCase().contains(filter))).collect(Collectors.toList());
-        List<ConfigPageButton> filteredBottom = bottomButtons.stream().filter(page -> !page.isHidden() && (filter.equals("") || name.toLowerCase().contains(filter) || page.name.toLowerCase().contains(filter) || page.description.toLowerCase().contains(filter))).collect(Collectors.toList());
+        //todo bugs: "Test Page" in test mod doesn't get filtered nicely
```
Comment (nxtdaydelivery) [inline src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java:59]: !!

File: src/main/java/cc/polyfrost/oneconfig/utils/JsonUtils.java [hunk 2]
```diff
@@ -39,6 +41,35 @@ public final class JsonUtils {
...
+    }
+
+    /**
+     * Checks whether the given string is a valid JSON.
+     * @param json The string to check.
+     * @return Whether the string is a valid JSON.
+     */
+    public static boolean isValid(String json) {
+        try {
+            PARSER.parse(json);
```
Comment (nxtdaydelivery) [inline src/main/java/cc/polyfrost/oneconfig/utils/JsonUtils.java:66]: ok but surely there is a lighter way to do this

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/EntityPlayerArgumentParser.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * The old player argument parser. Returns an {@link EntityPlayer}.
+ * @deprecated Use {@link PlayerArgumentParser} instead.
+ */
+@Deprecated
+public class EntityPlayerArgumentParser extends ArgumentParser<EntityPlayer> {
+    @Nullable
+    @Override
+    public EntityPlayer parse(@NotNull String arg) {
+        Deprecator.markDeprecated();
```
Comment (nxtdaydelivery) [inline versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/EntityPlayerArgumentParser.java:24]: no?

**Review by nxtdaydelivery — APPROVED** [review]
200 OK
read with a reasonable amount of confidence, but was a bit rushed towards the end.

File: src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java (hunk 1 above)
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java:59]: fixed

File: src/main/java/cc/polyfrost/oneconfig/utils/JsonUtils.java (hunk 2 above)
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/utils/JsonUtils.java:66]: wrong

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/EntityPlayerArgumentParser.java (hunk 3 above)
Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/EntityPlayerArgumentParser.java:24]: yes?

**Pull request merged**